### PR TITLE
Augment session events with cluster name

### LINKED ISSUE #3751: Provide Cluster Context in logs
### Feature Request
When aggregating and centralising Teleport logs, it's helpful to know which cluster the event happened on, without having to build extra tooling.  An example `session.start` event. 

**Example from `sudo tail -f /var/lib/teleport/log/events.log`**
```json
{"addr.local":"66.220.18.10:3022","addr.remote":"73.202.138.51:46798","code":"T2000I","ei":0,"event":"session.start","login":"root","namespace":"default","server_hostname":"teleport5auth","server_id":"a54859b4-0eba-4f36-af6a-452cb8379e38","server_labels":{"db_role":"master","db_type":"postgres"},"sid":"85eeb0a1-9aec-11ea-a0bd-125400432332","size":"80:25","time":"2020-05-20T22:52:04.111Z","uid":"b8bd5094-cb9f-4520-8f94-b71865fd9e19","user":"benarent"}
```
Formatted for easier analysis 
```json
{
   "addr.local":"66.220.18.10:3022",
   "addr.remote":"73.202.138.51:46798",
   "code":"T2000I",
   "ei":0,
   "event":"session.start",
   "login":"root",
   "namespace":"default",
   "server_hostname":"teleport5auth",
   "server_id":"a54859b4-0eba-4f36-af6a-452cb8379e38",
   "server_labels":{
      "db_role":"master",
      "db_type":"postgres"
   },
   "sid":"85eeb0a1-9aec-11ea-a0bd-125400432332",
   "size":"80:25",
   "time":"2020-05-20T22:52:04.111Z",
   "uid":"b8bd5094-cb9f-4520-8f94-b71865fd9e19",
   "user":"benarent"
}
```

sid = Session ID, can we used to retrieve session playback later, or share session. 

We should either provide `cluster_name` or proposed Alias https://github.com/gravitational/teleport/issues/3604. Since  `cluster_name` is immutable and baked in the SSH CA I would suggest adding that.

```diff
{
   "addr.local":"66.220.18.10:3022",
   "addr.remote":"73.202.138.51:46798",
   "code":"T2000I",
   "ei":0,
   "event":"session.start",
   "login":"root",
   "namespace":"default",
   "server_hostname":"teleport5auth",
   "server_id":"a54859b4-0eba-4f36-af6a-452cb8379e38",
   "server_labels":{
      "db_role":"master",
      "db_type":"postgres"
   },
   "sid":"85eeb0a1-9aec-11ea-a0bd-125400432332",
   "size":"80:25",
   "time":"2020-05-20T22:52:04.111Z",
   "uid":"b8bd5094-cb9f-4520-8f94-b71865fd9e19",
   "user":"benarent"
+   "cluster_name":"teleport-431"
}
```

### Motivation
It makes logs useful when aggregated across many clusters. 

### Who's it for?
OSS User, Pro, Enterprise

### PR DESCRIPTION
**Pull request by a-palchikov**
This PR augments session events with a cluster name.

Fixes https://github.com/gravitational/teleport/issues/3751.

### CODE REVIEW

File: lib/events/events.proto [hunk 1]
```diff
@@ -32,6 +32,9 @@ message Metadata {
 message SessionMetadata {
     // SessionID is a unique UUID of the session.
     string SessionID = 1 [ (gogoproto.jsontag) = "sid" ];
+
+    // ClusterName identifies the originating teleport cluster
+    string ClusterName = 2 [ (gogoproto.jsontag) = "cluster_name" ];
```
Comment (r0mant) [inline lib/events/events.proto:37]: I'm curious whether you've considered placing this in a separate `ClusterMetadata` struct? I don't know if cluster name really belongs to the session metadata, seems like a separate "class" of information to me but I might be misunderstanding.

Comment (russjones) [reply lib/events/events.proto:37]: What about `ServerMetadata`?

File: lib/events/events.proto [hunk 2]
```diff
@@ -61,6 +61,9 @@ message ServerMetadata {
     // session occurred on.
     map<string, string> ServerLabels = 5
         [ (gogoproto.nullable) = false, (gogoproto.jsontag) = "server_labels,omitempty" ];
+
+    // ClusterName identifies the originating teleport cluster
+    string ClusterName = 6 [ (gogoproto.jsontag) = "cluster_name" ];
```
Comment (awly) [inline lib/events/events.proto:66]: should this be in just `Metadata`?
aren't all events cluster-specific?

File: lib/srv/exec.go [hunk 3]
```diff
@@ -60,6 +60,8 @@ type ExecResult struct {
 
 // Exec executes an "exec" request.
 type Exec interface {
+	fmt.Stringer
```
Comment (awly) [inline lib/srv/exec.go:63]: no need to add it in the interface
as long as `remoteExec` implements `fmt.Stringer`, printing it with `%v` should pick it up

**Review by awly — APPROVED** [review]

**Comment by benarent** [comment]
We've a customer that's current aggregating all logs into Splunk for SIEM usage. Would we be able to backport this into 5.1

**Comment by a-palchikov** [comment]
@benarent Sure I can submit a backport.

**Pull request merged**